src/eventHandler.js

```diff
@@ -93,7 +93,20 @@ export default superclass => class extends superclass {
 			})
 		);
 
-		(eventRes || eventRes === null) && this._setTranslate([-pos, 0]);
+		if (eventRes || eventRes === null) {
+			this._setTranslate([-pos, 0]);
+		} else {
+			e.stop();
+			// release, animationEnd
+			touch.distance = 0;
+			touch.direction = direction || touch.direction;
```

Comment:
Why we should maintain direction here? Isn't it like reset everything kind of phase?